src/domain/users/users.repository.ts

```diff
@@ -46,4 +52,116 @@ export class UsersRepository implements IUsersRepository {
+        const walletRepository = entityManager.getRepository(Wallet);
+
+        const authenticatedWallet = await walletRepository.findOne({
+          where: { address: authPayload.signer_address },
+          relations: { user: true },
+        });
+
+        if (!authenticatedWallet?.user) {
+          throw new NotFoundException('User not found');
+        }
+
+        const wallets = await walletRepository.findBy({
+          user: authenticatedWallet.user,
+        });
+
+        return {
+          id: authenticatedWallet.user.id,
+          status: authenticatedWallet.user.status,
+          wallets: wallets.map((wallet) => ({
+            id: wallet.id,
+            address: wallet.address,
+          })),
+        };
+      },
+    );
+  }
+
+  async deleteUser(authPayload: AuthPayload): Promise<void> {
+    if (!authPayload.signer_address) {
+      throw new UnauthorizedException();
+    }
+
+    await this.postgresDatabaseService.transaction(
+      async (entityManager: EntityManager) => {
+        const walletRepository = entityManager.getRepository(Wallet);
+        const userRepository = entityManager.getRepository(DbUser);
+
+        const authenticatedWallet = await walletRepository.findOne({
+          where: { address: authPayload.signer_address },
+          relations: { user: true },
+        });
+
+        if (!authenticatedWallet?.user) {
+          throw new NotFoundException('User not found');
+        }
+
+        // TODO: Check that it also deletes all wallets of that user
+        await userRepository.delete(authenticatedWallet.user);
+      },
+    );
+  }
+
+  async deleteWalletFromUser(args: {
+    walletAddress: `0x${string}`;
+    authPayload: AuthPayload;
+  }): Promise<void> {
+    if (!args.authPayload.signer_address) {
+      throw new UnauthorizedException();
+    }
+
+    if (args.authPayload.signer_address === args.walletAddress) {
+      throw new ConflictException('Cannot remove the current wallet');
+    }
+
+    await this.postgresDatabaseService.transaction(
+      async (entityManager: EntityManager) => {
+        const walletRepository = entityManager.getRepository(Wallet);
+
+        const authenticatedWallet = await walletRepository.findOne({
+          where: { address: args.authPayload.signer_address },
+          relations: { user: true },
+        });
+
+        if (!authenticatedWallet?.user) {
+          throw new NotFoundException('User not found');
+        }
+
+        const userWalletCount = await walletRepository.count({
+          where: { user: authenticatedWallet.user },
+        });
+
+        if (userWalletCount <= 1) {
+          throw new BadRequestException(
+            'Cannot delete the last wallet of a user',
+          );
+        }
+
+        const deleteResult = await walletRepository.delete({
+          address: args.walletAddress,
+          user: authenticatedWallet.user,
+        });
+
+        if (!deleteResult.affected) {
```

Comment:
Nit: I would be more explicit here for reability. I initially thought this was a `boolean`.

```suggestion
        if (deleteResult.affected === 0) {
```

Reply:
The type of `affected` is `number | null`, so I’d leave it as is to account for both cases.